libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -2337,10 +2363,11 @@ string YulUtilFunctions::readFromStorageReferenceType(Type const& _type)
 		("memberMemoryOffset", structType.memoryOffsetOfMember(structMembers[i].name).str())
 		("memberSlotDiff",  memberSlotDiff.str())
 		("memberStorageOffset", to_string(memberStorageOffset))
-		("readFromStorage",
+		(
+			"readFromStorage",
 			structMembers[i].type->isValueType() ?
-				readFromStorageDynamic(*structMembers[i].type, true) :
-				readFromStorage(*structMembers[i].type, memberStorageOffset, true)
+			readFromStorageDynamic(*structMembers[i].type, true) :
```

Comment:
Actually I realized that it's the other way round: We can profit from the static read only for value types, since offset is zero for non-value types anyway. So Could you change this such that it directly calls `YulUtilFunctions::readFromStorageValueType` with a static offset?

Reply:
True that. If I just call `readFromStorage` in both cases, it should call `readFromStorageValueType` for value types. So I think it is better now.